server/src/main/scala/com/cloudera/livy/utils/LineBufferedStream.scala

```diff
@@ -36,17 +38,17 @@ class LineBufferedStream(inputStream: InputStream) extends Logging {
   private val thread = new Thread {
     override def run() = {
       val lines = Source.fromInputStream(inputStream).getLines()
-      for (line <- lines) {
+       for (line <- lines) {
         _lock.lock()
         try {
-          _lines = _lines :+ line
+          info(s"stdout: $line")
```

Comment:
@jerryshao In client mode we were seeing spark driver logs in livy log after great delay as the log statement was after the loop.  I agree log inside the lock is heavy. 

Let me know if there is a better way to handle the scenario.